tests/hp/field_transfer_05.cc

```diff
@@ -0,0 +1,145 @@
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+// Check that we can use FieldTransfer when multiple cells are changed from
+// FE_Nothing to FE_Q
+
+#include <deal.II/distributed/field_transfer.h>
+
+#include <deal.II/dofs/dof_handler.h>
+#include <deal.II/dofs/dof_tools.h>
+
+#include <deal.II/fe/fe_nothing.h>
+#include <deal.II/fe/fe_q.h>
+
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/tria.h>
+#include <deal.II/grid/tria_iterator.h>
+
+#include <deal.II/hp/fe_collection.h>
+
+#include <deal.II/numerics/vector_tools.h>
+
+#include "../tests.h"
+
+
+class TestFunction : public Function<2>
+{
+public:
+  double
+  value(const dealii::Point<2> &p, const unsigned int component) const override
+  {
+    return std::exp(10 * p[1]);
+  }
+};
+
+int
+main(int argc, char *argv[])
+{
+  Utilities::MPI::MPI_InitFinalize mpi_initialization(argc, argv, 1);
+
+  MPILogInitAll mpi_log;
+
+  parallel::distributed::Triangulation<2> triangulation(MPI_COMM_WORLD);
+  GridGenerator::hyper_cube(triangulation);
+  triangulation.refine_global(3);
+
+  hp::FECollection<2> fe_collection;
+  fe_collection.push_back(FE_Q<2>(1));
+  fe_collection.push_back(FE_Nothing<2>());
+
+  DoFHandler<2> dof_handler(triangulation);
+
+  // Assign FE_Nothing to half of the domain
+  for (auto cell : dof_handler.active_cell_iterators())
+    {
+      if (cell->is_locally_owned())
+        {
+          if (cell->center()[0] < 0.5)
+            {
+              cell->set_active_fe_index(0);
+            }
+          else
+            {
+              cell->set_active_fe_index(1);
+            }
+        }
+    }
+
+  dof_handler.distribute_dofs(fe_collection);
+
+  // Initialize solution
+  auto locally_relevant_dofs =
+    DoFTools::extract_locally_relevant_dofs(dof_handler);
+  LinearAlgebra::distributed::Vector<double> solution(
+    dof_handler.locally_owned_dofs(), locally_relevant_dofs, MPI_COMM_WORLD);
+
+  // Initialize with a non-constant function
+  VectorTools::interpolate(dof_handler, TestFunction(), solution);
+
+  {
+    deallog << "solution before: " << std::endl;
+    std::stringstream ss;
+    solution.print(ss);
+    deallog << ss.str() << std::endl;
+  }
+
+  parallel::distributed::experimental::
+    FieldTransfer<2, LinearAlgebra::distributed::Vector<double>>
+      field_transfer(dof_handler);
+  // Assign FE_Q to all the cells
+  for (auto cell : dof_handler.active_cell_iterators())
+    {
+      if (cell->is_locally_owned())
+        {
+          cell->set_future_fe_index(0);
+        }
+    }
+
+  // coarsen some cells
+  for (auto cell : dof_handler.active_cell_iterators())
+    {
+      if (cell->is_locally_owned())
+        {
+          if (cell->center()[0] < 0.5)
```

Comment:
Can you add a test where you coarsen FE_Nothing cells

Reply:
I added it but now I get an error (floating point exception in `LAPACKFullMatrix::vmult` inside `interpolate`) when trying to resolve the hanging node constraints.

Reply:
OK it is a weird thing to do anyway. I would be fine merging this PR which improves the current situation, if you open an issue that describes the problem.

Reply:
See #14807